x/rollapp/types/rollapp.go

```diff
@@ -28,31 +56,94 @@ func (r Rollapp) ValidateBasic() error {
 		return err
 	}
 
-	if r.MaxSequencers > MaxAllowedSequencers {
-		return errorsmod.Wrapf(ErrInvalidMaxSequencers, "max sequencers: %d, max sequencers allowed: %d", r.GetMaxSequencers(), MaxAllowedSequencers)
-	}
-	if uint64(len(r.PermissionedAddresses)) > r.GetMaxSequencers() {
-		return errorsmod.Wrapf(ErrTooManyPermissionedAddresses, "permissioned addresses: %d, max sequencers: %d", len(r.PermissionedAddresses), r.GetMaxSequencers())
-	}
-
-	// verifies that there's no duplicate address in PermissionedAddresses
-	// and addresses are in Bech32 format
-	permissionedAddresses := r.GetPermissionedAddresses()
-	if len(permissionedAddresses) > 0 {
-		duplicateAddresses := make(map[string]bool)
-		for _, item := range permissionedAddresses {
-			// check if the item/element exist in the duplicateAddresses map
-			_, exist := duplicateAddresses[item]
-			if exist {
-				return errorsmod.Wrapf(ErrPermissionedAddressesDuplicate, "address: %s", item)
-			}
-			// check Bech32 format
-			if _, err := sdk.AccAddressFromBech32(item); err != nil {
-				return errorsmod.Wrapf(ErrInvalidPermissionedAddress, "%s", err)
-			}
-			// mark as exist
-			duplicateAddresses[item] = true
-		}
+	_, err = sdk.AccAddressFromBech32(r.InitialSequencerAddress)
+	if err != nil {
+		return errorsmod.Wrap(ErrInvalidInitialSequencerAddress, err.Error())
+	}
+
+	if err = validateBech32Prefix(r.Bech32Prefix); err != nil {
+		return errorsmod.Wrap(ErrInvalidBech32Prefix, err.Error())
+	}
+
+	if r.GenesisChecksum == "" {
+		return errorsmod.Wrap(ErrEmptyGenesisChecksum, "GenesisChecksum")
+	}
+
+	if l := len(r.Alias); l == 0 || l > maxAliasLength {
+		return ErrInvalidAlias
+	}
+
+	if err = validateMetadata(r.Metadata); err != nil {
+		return errorsmod.Wrap(ErrInvalidMetadata, err.Error())
+	}
+
+	return nil
+}
+
+func validateBech32Prefix(prefix string) error {
+	bechAddr, err := sdk.Bech32ifyAddressBytes(prefix, sample.Acc())
+	if err != nil {
+		return err
+	}
+
+	bAddr, err := sdk.GetFromBech32(bechAddr, prefix)
+	if err != nil {
+		return err
+	}
+
+	if err = sdk.VerifyAddressFormat(bAddr); err != nil {
+		return err
+	}
+	return nil
+}
+
+func validateMetadata(metadata *RollappMetadata) error {
+	if metadata == nil {
+		return nil
+	}
+
+	if _, err := url.Parse(metadata.Website); err != nil {
+		return errorsmod.Wrap(ErrInvalidWebsiteURL, err.Error())
+	}
+
+	if len(metadata.Description) > maxDescriptionLength {
+		return ErrInvalidDescription
+	}
+
+	if err := validateBaseURI(metadata.LogoDataUri); err != nil {
+		return errorsmod.Wrap(ErrInvalidLogoURI, err.Error())
+	}
+
+	if err := validateBaseURI(metadata.TokenLogoUri); err != nil {
+		return errorsmod.Wrap(ErrInvalidTokenLogoURI, err.Error())
+	}
+
+	return nil
+}
+
+func validateBaseURI(dataURI string) error {
```

Comment:
missing UT

Reply:
it should be all covered [here](https://github.com/dymensionxyz/dymension/blob/2f822e8a450973b63702ea622e2970485bb8d8d4/x/rollapp/types/message_create_rollapp_test.go#L14)